### PR TITLE
Log Kueue's feature gates at the start-up

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
To make debugging easier for users

#### Special notes for your reviewer:
Tested e2e manually

#### Does this PR introduce a user-facing change?

```release-note
Users can now see configuration of Kueue's feature gates in `kueue-controller-manager` logs
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -121,3 +122,13 @@ func Enabled(f featuregate.Feature) bool {
...
 }
+
+func LogFeatureGates(log logr.Logger) {
+	features := make(map[featuregate.Feature]bool, len(defaultFeatureGates))
+	for f := range utilfeature.DefaultMutableFeatureGate.GetAll() {
+		if _, ok := defaultFeatureGates[f]; ok {
+			features[f] = Enabled(f)
+		}
+	}
+	log.V(2).Info("Feature gates:", "Feature gates:", features)
```
Comment (alculquicondor) [inline pkg/features/kube_features.go:133]:
```suggestion
	log.V(2).Info("Loaded feature gates", "featureGates", features)
```

**Review by tenzen-y — COMMENTED** [review]
Thank you.
/lgtm
/approve

**Pull request merged**